src/events/etcd.py

```diff
@@ -368,19 +416,52 @@ def _on_secret_changed(self, event: ops.SecretChangedEvent) -> None:
         if not self.charm.unit.is_leader():
             return
 
-        if self.charm.state.cluster.is_restore_in_progress:
-            logger.warning("Cannot update credentials while database restore is in progress.")
+        if (
+            self.charm.state.cluster.is_restore_in_progress
+            or self.charm.state.cluster.rebuild_cluster_in_progress
+        ):
+            logger.warning("Cannot update credentials while another operation is in progress.")
             event.defer()
             return
 
         if admin_secret_id := self.charm.config.get(INTERNAL_USER_PASSWORD_CONFIG):
             if admin_secret_id == event.secret.id:
                 self.update_admin_password(admin_secret_id)
 
+    def _on_rebuild_cluster_action(self, event: ActionEvent) -> None:
+        """Recover from majority failure by rebuilding the cluster membership configuration."""
+        if error := self._exists_preventing_reason():
```

Comment:
We should probably rename this one and the method used in the restore flow to be more specific, perhaps: 
```suggestion
        if error := self._exists_rebuild_preventing_reason():
```

Reply:
has been renamed